### PR TITLE
Text regression

### PR DESCRIPTION
**Pull request by qinyiwei**
Add text regression task. Implement three metrics: SegKtauCorr, SysPearsonCorr, RootMeanSquareError.

### CODE REVIEW

**Comment by pfliu-nlp** [comment]
Hey, @qinyiwei  thanks a lot for the PR. It would be nice if you could fix these continuous integration errors based on corresponding suggestions.

**Comment by neubig** [comment]
To make it easier to fix those errors, I'd suggest running `pre-commit install`, which will install pre-commit checks for formatting and type-checking. I think these things should be easy to check.

**Comment by pfliu-nlp** [comment]
These conflicts could be resolved manually.

File: explainaboard/loaders/text_regression.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                    FileLoaderField(5, target_field_names[5], str),
+                    FileLoaderField(6, target_field_names[6], str),
+                    FileLoaderField(7, target_field_names[7], str),
+                ],
+            ),
+        }
+
+    @classmethod
+    def default_output_file_loaders(cls) -> dict[FileType, FileLoader]:
+        field_name = [
```
Comment (pfliu-nlp) [inline explainaboard/loaders/text_regression.py:53]: To make the naming format consistent with others, we can lowercase these names and use "_" to connect two tokens if needed. For example `TestSet` -> `test_set`

File: explainaboard/tasks.py [hunk 2]
```diff
@@ -157,6 +157,16 @@ class TaskCategory:
             )
         ],
     ),
+    TaskCategory(
+        "text-regression",
+        "evaluate evaluation metrics.  ",
+        [
+            Task(
+                name=TaskType.text_regression,
```
Comment (pfliu-nlp) [inline explainaboard/tasks.py:197]: we could re-consider the task name, some other options, such as `meta-evaluation`

File: docs/task_text_regression.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+We have an example dataset file:
+* [data.tsv](./data/system_outputs/text_regression/wmt20-DA/cs-en/data.tsv)
+
+More dataset files can be found at [WMT-DA-20](https://drive.google.com/drive/u/0/folders/1JXpo0yxPLYlNgLbOfP1bzs9z6SOx76Wo).
+
+In order to perform analysis of your results, your system outputs should add one column of system score at the base of dataset file:
+
+```
+SYSName \t SEGID \t TestSet \t src \t ref \t sys \t manualRaw \t manualZ \t systemScore
```
Comment (pfliu-nlp) [inline docs/task_text_regression.md:19]: maybe we can explain a little bit about each column

**Comment by pfliu-nlp** [comment]
hi, @qinyiwei
>"explainaboard   --task text-regression --custom_dataset_paths ./data/system_outputs/text_regression/wmt20-DA/cs-en/data.tsv --system_outputs ./data/system_outputs/text_regression/wmt20-DA/cs-en/score.tsv --output_file_type tsv --output_dir output/cs-en --source_language en --target_language en --metrics SysPearsonCorr SegKtauCorr RootMeanSquareError"

This seems cannot work, detailed logs:

d5e73b377fe653710aeb05ae10826b0

907bedb3fffb593b5ff8ca8a1d0befd

**Review by neubig — CHANGES_REQUESTED** [review]
Thanks a lot for doing this!

My biggest comment is that I think we should rename the task as `nlg-meta-evaluation`, as @pfliu-nlp mentioned. When I see `text-regression`, I would expect just one textual input and one continuous-valued output. For example, we could implement this with `sst5` 5-level sentiment analysis (in a future PR).

Once the name is changed I'll do a more complete review!

**Comment by pfliu-nlp** [comment]
Hi, @neubig just re-named it.

File: explainaboard/metrics/regression.py [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+import numpy as np
+from scipy.stats import pearsonr
+
+from explainaboard.metrics.metric import Metric, MetricConfig, MetricStats
+from explainaboard.metrics.registry import register_metric_config
+
+
+@dataclass
+@register_metric_config
+class SegKtauCorrConfig(MetricConfig):
```
Comment (neubig) [inline explainaboard/metrics/regression.py:15]: (1) Could `no_human` and `threshold` be documented?
(2) Maybe we should consider the default for `threshold` being `0`? `25` seems a bit arbitrary and opaque.
(3) It'd be nicer if we could avoid having `Seg` be part of the class name, and rather define `corr_level = 'seg'` or `corr_level = 'sys'` as a variable in the configuration. That would make it easier to calculate `Ktau`, `Pearson`, `Spearman` etc. with respect to either segment-level or system-level correlation.

File: explainaboard/metrics/regression.py [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+                        conc += 1
+                    else:
+                        disc += 1
+                    num += 1
+                else:  # system i is worse than system j
+                    if score[i][1] < score[j][1]:
+                        conc += 1
+                    else:
+                        disc += 1
+                    num += 1
```
Comment (neubig) [inline explainaboard/metrics/regression.py:112]: The following suggestion is just to make things a bit more concise:
```suggestion
                if (abs(score[i][0] - score[j][0]) >= config.threshold):
                    manual_better = score[i][0] > score[j][0]
                    system_better = score[i][1] > score[j][1]
                    if manual_better == system_better:
                        conc += 1
                    else:
                        disc += 1
                    num += 1
```

**Review by neubig — COMMENTED** [review]
Thanks for the quick action, and overall looking pretty good! I have a few comments.

First, we can probably move the directory:
`data/system_outputs/text_regression`
->
`data/system_outputs/nlg_meta_evaluation`

File: explainaboard/metrics/regression.py (hunk 4 above)
Comment (pfliu-nlp) [reply explainaboard/metrics/regression.py:16]: fixed: `Regarding (1) (2) will fix them once I get feedbacks from Yiwei`

**Review by neubig — APPROVED** [review]
I approve now.

**Review by pfliu-nlp — APPROVED** [review]

**Pull request merged**